### PR TITLE
prevent tweets on cfps that open and close on the same date

### LINKED ISSUE #147: Shouldn't tweet if CFP has same open and close date
image

### PR DESCRIPTION
**Pull request by jwalton512**
closes #147

### CODE REVIEW

**Comment by jwalton512** [comment]
@mattstauffer noticed php-cs-fixer was a little greedier than i thought.  let me know if you want me to reset everything and reapply the changes.

File: tests/Console/Commands/TweetImportantCFPDatesTest.php [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+
+	public function setUp()
+	{
+		parent::setUp();
+
+		$this->twitter = m::mock(Twitter::class);
+	}
+
+	/** @test */
+	public function cfps_opening_or_closing_today_should_be_tweeted()
```
Comment (mattstauffer) [inline tests/Console/Commands/TweetImportantCFPDatesTest.php]: This test is a bit confusing. Would you be up to write a few more to show why we're getting two tweets here? Maybe one test for opening and another for closing?

Thanks!

File: app/Console/Commands/TweetImportantCFPDates.php [hunk 2]
```diff
@@ -14,9 +14,10 @@ class TweetImportantCFPDates extends Command
 
     protected $description = 'Send out a tweet with important upcoming CFP dates.';
 
-    public function __construct(Twitter $twitter)
+    public function __construct(Twitter $twitter, $tweetWaitSeconds = 30)
```
Comment (mattstauffer) [inline app/Console/Commands/TweetImportantCFPDates.php]: Nice refactor :+1:

File: app/Console/Commands/TweetImportantCFPDates.php [hunk 3]
```diff
@@ -94,12 +95,29 @@ private function tweet($message)
...
     }
+
+    /**
+     * Retrieve only those cfps which should be tweeted.
+     *
+     * @param \Illuminate\Database\Eloquent\Collection $cfps
+     *
+     * @return \Illuminate\Database\Eloquent\Collection
+     */
+    private function tweetable($cfps)
```
Comment (mattstauffer) [inline app/Console/Commands/TweetImportantCFPDates.php]: Maybe call this `$conferences` to be clearer? And pass `$conference` into the reject callback, just so it's easier to reason with what we're working with.

Great work on this.

File: tests/Console/Commands/TweetImportantCFPDatesTest.php [hunk 4]
```diff
@@ -0,0 +1,47 @@
+<?php
+
+use Carbon\Carbon;
+use Laracasts\TestDummy\Factory;
+use Mockery as m;
+use Symposium\Console\Commands\TweetImportantCFPDates;
+use Thujohn\Twitter\Twitter;
+
+class TweetImportantCFPDatesTest extends IntegrationTestCase
```
Comment (mattstauffer) [inline tests/Console/Commands/TweetImportantCFPDatesTest.php]: Could you convert this file to spaces?

**Comment by mattstauffer** [comment]
@jwalton512 Thanks! If you make these changes I think we'll be good.
This'll help a lot for addressing this: 

https://twitter.com/symposiumtweets/status/673492538670600192

**Comment by jwalton512** [comment]
@mattstauffer hopefully that looks better.  added some more tests, and tried to explain the conditions that were causing/preventing tweeting.

**Comment by jwalton512** [comment]
@mattstauffer another edge case that might need consideration (maybe already has), a conference where the cfp opens today and closes tomorrow would be tweeted twice.  is that desired?

**Comment by mattstauffer** [comment]
@jwalton512 Thanks! And no worries about the one-day-apart concept, I think that's less of a concern; but there's a glitch in how we're getting data from Joind.in that is making this same-day thing happen frequently.
